src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -68,7 +68,7 @@ def make_vcs_requirement_url(repo_url, rev, project_name, subdir=None):
       repo_url: the remote VCS url, with any needed VCS prefix (e.g. "git+").
       project_name: the (unescaped) project name.
     """
-    egg_project_name = pkg_resources.to_filename(project_name)
+    egg_project_name = canonicalize_name(project_name).replace("-", "_")
```

Comment:
Any particular reason to canonicalize the name here ? pkg_resources.to_filename does not do it.

Reply:
Removed.